ci/remote-watch.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+"""
+This command must be run in a git repository.
+
+It watches the remote branch for changes, killing the given PID when it detects
+that the remote branch no longer points to the local commit.
+
+If the commit message contains a line saying "CI_KEEP_ALIVE", then killing
+will not occur until the branch is deleted from the remote.
+
+If no PID is given, then the entire process group of this process is killed.
+"""
+
+# Prefer to keep this file Python 2-compatible so that it can easily run early
+# in the CI process on any system.
+
+import argparse
+import errno
+import logging
+import os
+import re
+import signal
+import subprocess
+import sys
+import time
+
+logger = logging.getLogger(__name__)
+
+GITHUB = "GitHub"
+TRAVIS = "Travis"
+
+
+def git(*args):
+    cmdline = ["git"] + list(args)
+    return subprocess.check_output(cmdline).decode("utf-8").rstrip()
+
+
+def get_current_ci():
+    if "GITHUB_WORKFLOW" in os.environ:
+        return GITHUB
+    elif "TRAVIS" in os.environ:
+        return TRAVIS
+    return None
+
+
+def get_ci_event_name():
+    ci = get_current_ci()
+    if ci == GITHUB:
+        return os.environ["GITHUB_EVENT_NAME"]
+    elif ci == TRAVIS:
+        return os.environ["TRAVIS_EVENT_TYPE"]
+    return None
+
+
+def get_repo_slug():
+    ci = get_current_ci()
+    if ci == GITHUB:
+        return os.environ["GITHUB_REPOSITORY"]
+    elif ci == TRAVIS:
+        return os.environ["TRAVIS_REPO_SLUG"]
+    return None
+
+
+def git_branch_info():
```

Comment:
can we make this function simpler?

Reply:
I can try again, but it might take a while... I recall it was originally simple and I had to complicate it to handle all the different cases.

Reply:
I don't think I know how to simplify this :\ I recall it was extremely painful getting all the edge cases right... Travis vs. GitHub, merge commit vs. normal commit, single commit vs. multiple commits, etc.